sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceManager.scala

```diff
@@ -81,3 +93,60 @@ class DataSourceManager extends Logging {
     manager
   }
 }
+
+/**
+ * Data Source V2 wrapper for Python Data Source.
+ */
+class PythonTableProvider(shortName: String) extends TableProvider {
+  private var sourceDataFrame: DataFrame = _
+
+  private def getOrCreateSourceDataFrame(
+      options: CaseInsensitiveStringMap, maybeSchema: Option[StructType]): DataFrame = {
+    if (sourceDataFrame != null) return sourceDataFrame
+    // TODO(SPARK-45600): should be session-based.
```

Comment:
This one should be fixed?

Reply:
I think so. The thing is that we should take a look into session inheritance, testcase, etc. So I leave this as a todo for now.